temporalio/lib/temporalio/worker.rb

```diff
@@ -435,6 +442,10 @@ def initialize(
           workflow_failure_exception_types:, workflow_definitions:
         )
 
+      # Convert deprecated max concurrent polls to poller behaviors if not specified
+      workflow_task_poller_behavior ||= SimpleMaximumPollerBehavior.new(maximum: max_concurrent_workflow_task_polls)
+      activity_task_poller_behavior ||= SimpleMaximumPollerBehavior.new(maximum: max_concurrent_activity_task_polls)
```

Comment:
I'm thinking we should set these as parameter defaults. Ruby is cool in that not only does the default expression not get evaluated if a value is provided, but a default parameter expression can use parameters defined before it.